pkg/forwarder/forwarder.go

```diff
@@ -0,0 +1,389 @@
+		fmt.Sprintf(`form-data; name="%s"; filename="%s"`,
+			escapeQuotes(fieldname), escapeQuotes(filename)))
+	h.Set("Content-Type", mimeType)
+	return w.CreatePart(h)
+}
+
+func (fm *ForwardManager) buildRequest(doc string, filename string, status validationStatus, target *target) (*http.Request, error) {
+	body := new(bytes.Buffer)
+	writer := multipart.NewWriter(body)
+
+	var err error
+	part := func(name, fname, mimeType, content string) {
+		if err != nil {
+			return
+		}
+		if fname == "" {
+			err = writer.WriteField(name, content)
+			return
+		}
+		var w io.Writer
+		if w, err = createFormFile(writer, name, fname, mimeType); err == nil {
+			_, err = w.Write([]byte(content))
+		}
+	}
+
+	base := filepath.Base(filename)
+	part("advisory", base, "application/json", doc)
+	part("validation_status", "", "text/plain", string(status))
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := writer.Close(); err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, target.url, body)
+	if err != nil {
+		return nil, err
+	}
+	contentType := writer.FormDataContentType()
+	if target.header != nil {
+		req.Header = target.header.Clone()
+	}
+	req.Header.Set("Content-Type", contentType)
+	return req, nil
+}
+
+func (fm *ForwardManager) loadDocument(ctx context.Context, documentID int64) ([]byte, string, error) {
+	expr := query.FieldEqInt("id", documentID)
+
+	fields := []string{"original", "filename"}
+	builder := query.SQLBuilder{}
+	builder.CreateWhere(expr)
+	sql := builder.CreateQuery(fields, "", -1, -1)
+
+	var original []byte
+	var filename string
+
+	if err := fm.db.Run(
+		ctx,
+		func(rctx context.Context, conn *pgxpool.Conn) error {
+			return conn.QueryRow(rctx, sql, builder.Replacements...).Scan(&original, &filename)
+		}, 0,
+	); err != nil {
+		return nil, "", err
+	}
+	return original, filename, nil
+}
+
+func (fm *ForwardManager) fetchNewDocuments(ctx context.Context, url string, publisher *string) ([]int64, error) {
+	var wherePublisher string
+	args := []any{url}
+	if publisher != nil {
+		wherePublisher = "publisher = $2 AND "
+		args = append(args, *publisher)
+	}
+	var documentIDs []int64
+
+	if err := fm.db.Run(
+		ctx,
+		func(rctx context.Context, conn *pgxpool.Conn) error {
+			fetchSQL := `SELECT documents_id FROM events_log ` +
+				`WHERE documents_id in ` +
+				`(SELECT id FROM documents WHERE ` + wherePublisher +
+				`id NOT IN (SELECT documents_id FROM forwarded_documents WHERE url = $1)) ORDER BY time DESC`
+			rows, _ := conn.Query(rctx, fetchSQL, args...)
+			var err error
+			documentIDs, err = pgx.CollectRows(
+				rows,
+				func(row pgx.CollectableRow) (int64, error) {
+					var documentID int64
+					err := row.Scan(&documentID)
+					return documentID, err
+				})
+			return err
+		}, 0,
+	); err != nil {
+		return documentIDs, err
+	}
+
+	return documentIDs, nil
+}
+
+func (fm *ForwardManager) fetchValidationStatus(ctx context.Context, documentID int64) validationStatus {
```

Comment:
Could be moved into the statement in loadDocument. To do so just create the SQL statement with the  `sb.WhereClause` from the SQLBuilder and not `.CreateQuery`